app/main/views/briefs.py

```diff
@@ -143,6 +143,24 @@ def list_briefs():
             *(status.strip() for status in request.args['status'].split(','))
         )
 
+    status_filters = ['created', 'published', 'withdrawn', 'cancelled', 'unsuccessful', 'closed']
+    temporal_filters = ['on', 'before', 'after']
+    status_date_filters = [
+        '{}_{}'.format(status, temporal) for status in status_filters for temporal in temporal_filters
+    ]
+
+    for status_date_key in [arg for arg in request.args.keys() if arg in status_date_filters]:
+        status, temporal_filter = status_date_key.split('_')
+        status = "applications_closed" if status == "closed" else status  # Hack for 'closed' status
+        day_value = datetime.strptime(request.args.get(status_date_key), "%Y-%m-%d")
+        if temporal_filter == 'on':
+            day_end = datetime(day_value.year, day_value.month, day_value.day, 23, 59, 59)
+            briefs = briefs.has_date_field_between("{}_at".format(status), day_value, day_end, inclusive=True)
```

Comment:
Feels slightly weird that we are saying to inclusively include `23:59:59` given that it on the same day and not in a different time period but our method does require us to do this.

Not entirely sure if this is a good suggestion but you could instead pass the next day at `00:00:00` non inclusively in as the end value.

Reply:
I suppose this would cover an edge case of a `23:59:59.999999` timestamp! However our `has_datetime_field_between` doesn't currently allow separate inclusive flags for the lower and upper bounds. It's possible to chain the before/after query methods and use them separately, but I think this would be slightly less efficient than using the `sql_and` in the between method.

I'll update the range to include microseconds as a compromise.